### ISSUE TITLE
error socket.io close connection

### ISSUE THREAD

**Issue by arteck** [ioBroker/ioBroker.admin#1000]
after login 

![grafik](https://user-images.githubusercontent.com/6681528/121926025-cc58c180-cd3d-11eb-8e5b-08dba3215c72.png)

**Comment by GermanBluefox** [comment]
Can happen.

**Comment by arteck** [comment]
jep https

**Comment by Feuer-sturm** [comment]
Admin 5.1.10


After enabling authentification (admin) without using https

```
2021-07-11 21:11:53.991 - info: admin.0 (31232) starting. Version 5.1.10 in /opt/iobroker/node_modules/iobroker.admin, node: v12.22.2, js-controller: 3.3.14
2021-07-11 21:11:54.087 - info: admin.0 (31232) requesting all states
2021-07-11 21:11:54.089 - info: admin.0 (31232) requesting all objects
2021-07-11 21:11:58.345 - info: admin.0 (31232) received all objects
2021-07-11 21:11:59.101 - info: admin.0 (31232) http server listening on port 8081
2021-07-11 21:11:59.104 - info: admin.0 (31232) Use link "http://localhost:8081" to configure.
2021-07-11 21:12:01.880 - error: admin.0 (31232) failed connection to socket.io from ::ffff:192.168.178.20:
2021-07-11 21:12:01.886 - error: admin.0 (31232) error
2021-07-11 21:12:01.971 - error: admin.0 (31232) failed connection to socket.io from ::ffff:192.168.178.20:
2021-07-11 21:12:01.973 - error: admin.0 (31232) error
```

**Comment by Feuer-sturm** [comment]
Admin 5.1.12

By opening the website http://192.168.178.77:8081/?login&href=#tab-intro the warning "cannot parse "undefined"" is shown in addition to the error:

2021-07-21 17:31:28.306  - warn: admin.0 (1315) Cannot parse "undefined"
2021-07-21 17:31:28.332  - error: admin.0 (1315) error
2021-07-21 17:31:28.406  - warn: admin.0 (1315) Cannot parse "undefined"

**Comment by ScumbagSteve** [comment]
I can confirm I got the same problem. 
When I open Admin5 with an iOS device wit latest stable updates (iPad or iPhone), I only see the iobroker logo (loading icon).
When I disable https, it's working like a charm.

Browser cache is cleared and I tried it with chrome and safari.
Could to be a problem with the iOS certificate handling, that would explain why both browsers are affected.
I disable https until the problem is fixed.

**Comment by Boronsbruder** [comment]
Admin v5.1.23
js-controller v3.3.15
socket.io v3.1.4

Same here, everytime the admin-webif(https) requests a new login (Firefox on Android, FF on Win10, Chrome on Win10)
```
2021-08-13 06:58:47.616  - error: admin.0 (24342) failed connection to socket.io from 192.168.116.99
2021-08-13 06:58:47.625  - error: admin.0 (24342) error
```

**Comment by Cyberfrog-Buchen** [comment]
Same problem. Everytime I sign in http everything works, but get error in log.

admin.0 | 2021-08-19 13:20:16.631 | error | error
admin.0 | 2021-08-19 13:20:16.630 | error | failed connection to socket.io from ::ffff:192.168.xxx.xxx:
admin.0 | 2021-08-19 13:20:16.523 | error | error
admin.0 | 2021-08-19 13:20:16.511 | error | failed connection to socket.io from ::ffff:192.168.xxx.xxx:
admin.0 | 2021-08-19 13:20:15.808 | warn | Cannot parse "undefined"

Node.js: v12.22.1
NPM: v6.14.12
Admin v5.1.23
js-controller v3.3.15

Any idea ?

**Comment by baerenwaldfreund** [comment]
Gleicher Fehler hier.

    Debian Debian 10.2.1-6
    Admin 5.1.28 (Authentication enabled, HTTPS enabled)
    js-controller 3.3.19
    nodejs v14.18.1
    npm 6.14.15


    




Der Fehler erscheint mit:

   Google Chrome Version 96.0.4664.45 (Offizieller Build) (64-Bit)
   Microsoft Edge Version 96.0.1054.29 (Offizielles Build) (64-Bit)
   Firefox 94.0.1 (64-bit)

`<html>
<body>


<html>
<body>

admin.0 | 2021-11-20 13:04:52.353 | error | error -- | -- | -- | --
admin.0 | 2021-11-20 13:04:52.349 | error | failed connection to socket.io from ::ffff:192.168.178.106:

</body>
</html>

</body>
</html>`

**Comment by Feuer-sturm** [comment]
Admin 5.1.28
js-controller 3.3.20

Mit dieser Konfiguration kommt im Log noch folgender Eintrag `admin.0 (2811) Cannot parse "undefined"`, wenn man folgende Schritte ausführt:

Schritte zum reproduzieren:
-  Anmelden mit Benutzername und Passwort
-  iobroker stop
-  iobroker start
- Admin wechselt zurück auf den Login Screen
- Fehler kommt im Log



```
25 Nov 2021 20:40:31.042 iobrokerTestsystem admin.0 (2811) error
25 Nov 2021 20:40:31.039 iobrokerTestsystem admin.0 (2811) Cannot parse "undefined"
25 Nov 2021 20:40:31.013 iobrokerTestsystem host.ioBrokerTestsystem instance system.adapter.hm-rega.0 started with pid 2867
25 Nov 2021 20:40:30.920 iobrokerTestsystem javascript.0 (2826) npm install xml2js --production (System call)
25 Nov 2021 20:40:30.850 iobrokerTestsystem admin.0 (2811) Cannot parse "undefined"
25 Nov 2021 20:40:30.722 iobrokerTestsystem javascript.0 (2826) starting. Version 5.2.13 in /opt/iobroker/node_modules/iobroker.javascript, node: v14.18.1, js-controller: 3.3.20
25 Nov 2021 20:40:30.624 iobrokerTestsystem admin.0 (2811) error
25 Nov 2021 20:40:30.610 iobrokerTestsystem admin.0 (2811) Cannot parse "undefined"
25 Nov 2021 20:40:29.347 iobrokerTestsystem scenes.0 (2833) starting. Version 2.3.8 in /opt/iobroker/node_modules/iobroker.scenes, node: v14.18.1, js-controller: 3.3.20
25 Nov 2021 20:40:27.074 iobrokerTestsystem host.ioBrokerTestsystem instance system.adapter.hm-rpc.0 started with pid 2844
25 Nov 2021 20:40:25.465 iobrokerTestsystem admin.0 (2811) Use link "http://localhost:8081" to configure.
25 Nov 2021 20:40:25.462 iobrokerTestsystem admin.0 (2811) http server listening on port 8081
25 Nov 2021 20:40:23.352 iobrokerTestsystem host.ioBrokerTestsystem instance system.adapter.scenes.0 started with pid 2833
25 Nov 2021 20:40:23.302 iobrokerTestsystem admin.0 (2811) received all objects
25 Nov 2021 20:40:21.589 iobrokerTestsystem host.ioBrokerTestsystem instance system.adapter.javascript.0 started with pid 2826
25 Nov 2021 20:40:18.205 iobrokerTestsystem admin.0 (2811) requesting all objects
25 Nov 2021 20:40:18.203 iobrokerTestsystem admin.0 (2811) requesting all states
25 Nov 2021 20:40:18.056 iobrokerTestsystem admin.0 (2811) starting. Version 5.1.28 in /opt/iobroker/node_modules/iobroker.admin, node: v14.18.1, js-controller: 3.3.20 
```

**Comment by Jade64** [comment]
Same here (if it helps)

Admin 5.2.3
js-controller 3.3.22

<html><body>



2022-02-04 12:27:08.854  - warn: admin.0 (718) Cannot parse "undefined"
2022-02-04 12:27:09.169  - error: admin.0 (718) failed connection to socket.io from ::ffff:192.168.1.45:
2022-02-04 12:27:09.173  - error: admin.0 (718) error
2022-02-04 12:27:09.197  - error: admin.0 (718) failed connection to socket.io from ::ffff:192.168.1.45:
2022-02-04 12:27:09.198  - error: admin.0 (718) error



</body>
</html>

**Comment by klein0r** [comment]
I don't get the error on v5.3.1 anymore... fixed? 😄

**Comment by Diginix** [comment]
@klein0r nope, I have 5.3.1 with SSL and if I restart iobroker and have an already openend admin tab in browser I still get this error.

**Comment by Kettenknecht** [comment]
I also get this error. I wonder why: 
1. it tries IPV6 instead of IPV4, because i can't find out any V6 address for socket.io via nslookup
2. it uses an V6 mapped V4 address, instead the native V6 address
3. it uses the mapped V4 address of the client computer (this from which i browsing), not the address of the computer where iobroker resides

**Comment by tmycgn** [comment]
Same here.

```
Raspbian GNU/Linux 10 (buster)
Admin 5.3.3
NPM 6.14.14
Node.js v14.19.1
js-controller 4.0.21
```

![image](https://user-images.githubusercontent.com/82310796/160273383-7474361b-65ba-479d-a50f-732418c2bff3.png)

**Comment by waldemarG007** [comment]
PLATTFORM: LINUX
BETRIEBSSYSTEM: LINUX
ARCHITEKTUR: ARM
CPUS: 4
GESCHWINDIGKEIT: 900 MHZ
MODELL: ARMV7 PROCESSOR REV 3 (V7L)
RAM: 1.8 GB
SYSTEM-BETRIEBSZEIT: 15:50:41
NODE.JS: V12.22.11
TIME: 1648462714070
TIMEOFFSET: -120
ADAPTER-ANZAHL: 406
NPM: 6.14.16
DATENTRÄGERGRÖSS: 57.4 GB
FREIER FESTPLATTENSPEICHER: 54.5 GB
BETRIEBSZEIT: 15:50:18
AKTIVE INSTANZEN: 10

admin.0 | 2022-03-28 09:33:20.458 | error | error
admin.0 | 2022-03-28 09:33:20.445 | error | failed connection to socket.io from ::ffff:192.168.178.28:
admin.0 | 2022-03-28 08:02:49.032 | error | error
admin.0 | 2022-03-28 08:02:49.002 | error | failed connection to socket.io from ::ffff:192.168.178.28:

**Comment by waldemarG007** [comment]
nach dem Update NODE.JS: V12.22.11 auf v14.19.1 ist dieses Fehler bei mir weg.

**Comment by Soulguard** [comment]
Same here, also with NODE.JS v14.19.1, but still iobroker works.

```
Admin 5.3.1
NPM 6.14.16
Node.js v14.19.1
js-controller 4.0.21
```
admin.0 | 2022-03-28 09:33:20.458 | error | error
admin.0 | 2022-03-28 09:33:20.445 | error | failed connection to socket.io from ::ffff:192.168.100.30:
admin.0 | 2022-03-28 08:02:49.032 | error | error
admin.0 | 2022-03-28 08:02:49.002 | error | failed connection to socket.io from ::ffff:192.168.100.30:

**Comment by Diginix** [comment]
Here the same with active ssl for admin. I guess it happens after an iobroker restart and already opened admin tab in browser.

**Comment by aortmannm** [comment]
Is it possible that the message always comes when the browser says that the certificate is not trusted? Seems to be the issue for me.

**Comment by Apollon77** [comment]
This could be

**Comment by Diginix** [comment]
Yes, I guess it has something to do with the enabled self signed ssl cert for admin.
For me it is no bug that has to be fixed if it is not that simple or nearly impossible.

**Comment by 150d** [comment]
Shouldn't be the case here: My browser fully trusts the certificate used by iobroker, and I'm still getting these errors.

I'm also already running nodejs v14 (same errors before with v12).

**Comment by jolu1705** [comment]
Same here

**Comment by MichaelUray** [comment]
I also have these messages in my logs.
Authentication is enabled in the admin module as well as in vis.
https is disabled since the encryption gets handled by a reverse proxy (Traefik) with valid let's encrypt certificates.

<html><body>


Quelle | Zeit | Debug | Nachricht
-- | -- | -- | --
admin.0 | 2022-04-24 16:41:06.499 | error | error
admin.0 | 2022-04-24 16:41:06.499 | error | failed connection to socket.io from ::ffff:172.18.0.10:
admin.0 | 2022-04-24 16:41:06.404 | error | error
admin.0 | 2022-04-24 16:41:06.401 | error | failed connection to socket.io from ::ffff:172.18.0.10:


</body>
</html>

**Comment by crycode-de** [comment]
This error message is also logged periodically if you have authentication enabled in ioBroker and you have the [ioBroker VSCode extension](https://marketplace.visualstudio.com/items?itemName=nokxs.iobroker-javascript) installed and VSCode running.
(The extension has no support for authentication at the moment.)

**Comment by iobFan** [comment]
Fehler tritt bei mir ebenfalls bei jedem Login auf:

<html>
<body>


admin.0 | 2022-06-19 10:49:14.670 | error | Error: "error"
-- | -- | -- | --
admin.0 | 2022-06-19 10:49:14.661 | info | failed connection to socket.io from xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxxxxx:


</body>
</html>

**Comment by GermanBluefox** [comment]
> failed connection

Admin version? 
Please test with 6.0.7

**Comment by arteck** [comment]
thx
